src/Data/Serialization.h fails to compile with Boost 1.58.0

The function `boost::serialization::stl::load_collection` no longer exists, neither do `archive_input_seq` nor `no_reserve_imp`

<pre>
/builddir/build/BUILD/IQmol-df94cf68b9d9d813e4d781c7edb9ac638724dadf/src/Data/Serialization.h: In function ‘void boost::serialization::load(Archive&, QList<T>&, unsigned int)’:
/builddir/build/BUILD/IQmol-df94cf68b9d9d813e4d781c7edb9ac638724dadf/src/Data/Serialization.h:170:4: error: ‘load_collection’ is not a member of ‘boost::serialization::stl’
    boost::serialization::stl::load_collection< 
    ^
/builddir/build/BUILD/IQmol-df94cf68b9d9d813e4d781c7edb9ac638724dadf/src/Data/Serialization.h:171:14: error: expected primary-expression before ‘,’ token
       Archive, 
              ^
/builddir/build/BUILD/IQmol-df94cf68b9d9d813e4d781c7edb9ac638724dadf/src/Data/Serialization.h:172:15: error: expected primary-expression before ‘,’ token
       QList<U>, 
               ^
/builddir/build/BUILD/IQmol-df94cf68b9d9d813e4d781c7edb9ac638724dadf/src/Data/Serialization.h:173:7: error: ‘archive_input_seq’ is not a member of ‘boost::serialization::stl’
       boost::serialization::stl::archive_input_seq<Archive, QList<U> >,
       ^
/builddir/build/BUILD/IQmol-df94cf68b9d9d813e4d781c7edb9ac638724dadf/src/Data/Serialization.h:173:59: error: expected primary-expression before ‘,’ token
       boost::serialization::stl::archive_input_seq<Archive, QList<U> >,
                                                           ^
/builddir/build/BUILD/IQmol-df94cf68b9d9d813e4d781c7edb9ac638724dadf/src/Data/Serialization.h:173:70: error: expected primary-expression before ‘>’ token
       boost::serialization::stl::archive_input_seq<Archive, QList<U> >,
                                                                      ^
/builddir/build/BUILD/IQmol-df94cf68b9d9d813e4d781c7edb9ac638724dadf/src/Data/Serialization.h:173:71: error: expected primary-expression before ‘,’ token
       boost::serialization::stl::archive_input_seq<Archive, QList<U> >,
                                                                       ^
/builddir/build/BUILD/IQmol-df94cf68b9d9d813e4d781c7edb9ac638724dadf/src/Data/Serialization.h:174:7: error: ‘no_reserve_imp’ is not a member of ‘boost::serialization::stl’
       boost::serialization::stl::no_reserve_imp< QList<U> > >(ar, t);
       ^
/builddir/build/BUILD/IQmol-df94cf68b9d9d813e4d781c7edb9ac638724dadf/src/Data/Serialization.h:174:59: error: expected primary-expression before ‘>’ token
       boost::serialization::stl::no_reserve_imp< QList<U> > >(ar, t);
                                                           ^
/builddir/build/BUILD/IQmol-df94cf68b9d9d813e4d781c7edb9ac638724dadf/src/Data/Serialization.h:174:61: error: expected primary-expression before ‘>’ token
       boost::serialization::stl::no_reserve_imp< QList<U> > >(ar, t);
                                                             ^
</pre>

--- a/src/Data/Serialization.h
+++ b/src/Data/Serialization.h
@@ -32,6 +32,11 @@
 #include <boost/archive/xml_oarchive.hpp>
 #include <boost/serialization/string.hpp>
 #include <boost/serialization/list.hpp>
+#include <boost/version.hpp>
+#if BOOST_VERSION >= 105800
+#include <boost/serialization/collections_save_imp.hpp>
+#include <boost/serialization/collections_load_imp.hpp>
+#endif
 #include <exception>
 #include "Matrix.h"
 

